### PR TITLE
feat(skia): [GTK][WPF][Tizen] Add `CoreWindow.[KeyUp|KeyDown]` and `UIElement.[KeyUP|KeyDown]`

### PR DESCRIPTION
**Pull request by jeromelaban**
GitHub Issue (If applicable): closes https://github.com/unoplatform/uno/issues/4318

## PR Type

What kind of change does this PR introduce?
- Feature

## What is the new behavior?

[GTK][WPF][Tizen] Add `CoreWindow.[KeyUp|KeyDown]` and `UIElement.[KeyUP|KeyDown]`

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI.Runtime.Skia.Gtk/GTK/GtkCoreWindowExtension.Keyboard.cs [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+using Windows.UI.Core;
+using Windows.UI.Input;
+
+namespace Uno.UI.Runtime.Skia
+{
+	public partial class GtkCoreWindowExtension : ICoreWindowExtension
+	{
+		partial void InitializeKeyboard()
+		{
+			Gtk.Key.SnooperInstall(OnKeySnoop);
```
Comment (davidjohnoliver) [inline src/Uno.UI.Runtime.Skia.Gtk/GTK/GtkCoreWindowExtension.Keyboard.cs:31]: 'Snooper' :D

File: src/Uno.UI.Runtime.Skia.Gtk/GTK/GtkCoreWindowExtension.Keyboard.cs [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+						{
+							ScanCode = evt.HardwareKeycode,
+							RepeatCount = 1,
+						}));
+			}
+			catch (Exception e)
+			{
+				if (this.Log().IsEnabled(LogLevel.Error))
+				{
+					this.Log().Error("Failed to raise KeyDown", e);
```
Comment (davidjohnoliver) [inline src/Uno.UI.Runtime.Skia.Gtk/GTK/GtkCoreWindowExtension.Keyboard.cs:73]: `Application.RaiseRecoverableUnhandledException()` ?

File: src/Uno.UI.Runtime.Skia.Gtk/GTK/GtkCoreWindowExtension.Keyboard.cs [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+						virtualKey,
+						new CorePhysicalKeyStatus
+						{
+							ScanCode = evt.HardwareKeycode,
+							RepeatCount = 1,
+						}));
+			}
+			catch(Exception e)
+			{
+				this.Log().Error("Failed to raise KeyUp", e);
```
Comment (davidjohnoliver) [inline src/Uno.UI.Runtime.Skia.Gtk/GTK/GtkCoreWindowExtension.Keyboard.cs:101]: ditto

File: src/Uno.UI.Runtime.Skia.Gtk/GTK/GtkCoreWindowExtension.Keyboard.cs [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+			}
+		}
+
+		private VirtualKey ConvertKey(Gdk.Key key)
+		{
+			return key switch
+			{
+				Gdk.Key.VoidSymbol => VirtualKey.None,
+
+				// Gdk.Key.Left => VirtualKey.LeftButton,
```
Comment (davidjohnoliver) [inline src/Uno.UI.Runtime.Skia.Gtk/GTK/GtkCoreWindowExtension.Keyboard.cs:111]: should the commented entries be committed? If so maybe add a comment indicating why

File: src/Uno.UI.Runtime.Skia.Gtk/GTK/GtkCoreWindowExtension.cs [hunk 5]
```diff
@@ -41,7 +41,7 @@ public void ReleasePointerCapture()
 		public void SetPointerCapture()
 			=> this.Log().Error("Pointer capture is not supported on GTK");
 
-		public GtkUIElementPointersSupport(object owner)
+		public GtkCoreWindowExtension(object owner)
```
Comment (davidjohnoliver) [inline src/Uno.UI.Runtime.Skia.Gtk/GTK/GtkCoreWindowExtension.cs:44]: Is this a binary breaking change?

File: src/Uno.UI.Runtime.Skia.Tizen/Tizen/TizenCoreWindowExtension.Keyboard.cs [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+			{
+				return (VirtualKey.None, (KeyCode)0);
+			}
+		}
+
+		private VirtualKey ConvertKey(object keyCode)
+		{
+			return keyCode switch
+			{
+				// KeyCode.LeftButton => VirtualKey.LeftButton,
```
Comment (davidjohnoliver) [inline src/Uno.UI.Runtime.Skia.Tizen/Tizen/TizenCoreWindowExtension.Keyboard.cs:106]: same point here

**Review by davidjohnoliver — APPROVED** [review]

File: src/Uno.UI.Runtime.Skia.Gtk/GTK/GtkCoreWindowExtension.cs (hunk 5 above)
Comment (jeromelaban) [reply src/Uno.UI.Runtime.Skia.Gtk/GTK/GtkCoreWindowExtension.cs:44]: There's no good reason for those types to be public, I'll change the others.

**Review by carldebilly — APPROVED** [review]

**Pull request merged**